src/opnsense/mvc/app/controllers/OPNsense/DHCP/Api/LeasesController.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+/*
+ * Copyright (C) 2023 Deciso B.V.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+ * AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+namespace OPNsense\DHCP\Api;
+
+use OPNsense\Base\ApiControllerBase;
+use OPNsense\Core\Backend;
+use OPNsense\Core\Config;
+use OPNsense\Firewall\Util;
+
+class LeasesController extends ApiControllerBase
+{
+    public function searchLeaseAction()
+    {
+        $this->sessionClose();
+        $inactive = $this->request->get('inactive');
+        $selected_interfaces = $this->request->get('selected_interfaces');
+        $backend = new Backend();
+        $config = Config::getInstance()->object();
+        $online = [];
+
+        /* get ARP data to match on */
+        $arp_data = json_decode($backend->configdRun('interface list arp json'), true);
+
+        foreach ($arp_data as $arp_entry) {
+            if (!$arp_entry['expired']) {
+                array_push($online, $arp_entry['mac'], $arp_entry['ip']);
+            }
+        }
+
+        /* get configured static leases */
+        $sleases = json_decode($backend->configdRun('dhcpd list static'), true);
+
+        /* include inactive leases if requested */
+        $leases = json_decode($backend->configdpRun('dhcpd list leases', [$inactive]), true);
+        foreach ($leases as $idx => $lease) {
+            /* set defaults */
+            $leases[$idx]['type'] = 'dynamic';
```

Comment:
@AdSchellevis should `type` and `state` be wrapped in a `gettext()`?

Reply:
@swhite2 a bit late to the party, small issue with my mail, but I don't mind keeping it untranslated for now, from an API perspective its better to keep values untranslated.